torchgeo/datasets/geo.py

```diff
@@ -89,8 +111,55 @@ def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
             IndexError: if query is not found in the index
         """
 
-    def __add__(self, other: "GeoDataset") -> "ZipDataset":  # type: ignore[override]
-        """Merge two GeoDatasets.
+    def __add__(  # type: ignore[override]
+        self, other: "GeoDataset"
+    ) -> "IntersectionDataset":
+        """Take the addition of two GeoDatasets.
+
+        :class:`GeoDataset` addition can be ambiguous and is no longer supported.
+        Users should instead use the intersection or union operator.
+
+        Args:
+            other: another dataset
+
+        Returns:
+            a single dataset
+
+        Raises:
+            ValueError: if other is not a :class:`GeoDataset`
+
+        Warns:
+            DeprecationWarning: if addition is used instead of intersection or union
+
+        .. deprecated: 0.2
+           Use the intersection or union operator instead.
+        """
+        # TODO: in the next release, we should keep this function but instead raise
+        # NotImplementedError. Otherwise, this will return a ConcatDataset that won't
+        # function properly without any warnings to the user.
```

Comment:
Didn't want to remove this all of a sudden without deprecating it first, but let me know if you want me to just remove it. We could/should also do the same for `ZipDataset`.